tokenizer_ts/src/bytePairEncode.ts

```diff
@@ -93,45 +106,35 @@ export function bytePairEncode(
         return rank;
       }
     }
-    return Number.MAX_SAFE_INTEGER;
+    return maxRank;
   }
 
-  for (let i = 0; i < byteIndicesAndRanks.length - 2; i++) {
-    const rank = getRank(i);
-    if (rank !== Number.MAX_SAFE_INTEGER) {
-      byteIndicesAndRanks[i][1] = rank;
+  while (minRank !== maxRank) {
+    byteIndicesAndRanks[minIndex][1] = getRank(minIndex, 1);
+    if (minIndex > 0) {
+      byteIndicesAndRanks[minIndex - 1][1] = getRank(minIndex - 1, 1);
     }
-  }
+    byteIndicesAndRanks.splice(minIndex + 1, 1);
+
 
-  while (byteIndicesAndRanks.length > 1) {
-    let minRank: [number, number] = [0, Number.MAX_SAFE_INTEGER];
```

Comment:
did you separate the minRank into two different integers because it's more efficient than re-assigning an array like this?

Reply:
yep exactly. Less garbage for collection.